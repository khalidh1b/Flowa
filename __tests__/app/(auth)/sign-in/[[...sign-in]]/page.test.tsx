import { render, screen } from "@testing-library/react";
import SignInPage from "@/app/(auth)/sign-in/[[...sign-in]]/page";
import React from "react";

jest.mock("@clerk/nextjs", () => ({
    SignIn: () => React.createElement("div", null, "mock signin component"),
}));

describe("SignInPage", () => {
    
    //unit test
    it("renders the signin component", () => {
        render(<SignInPage/>)
        expect(screen.getByText("mock signin component")).toBeInTheDocument()
    })

    //snapshot test
    it("matches snapshot", () => {
        const { asFragment } = render(<SignInPage/>)
        expect(asFragment()).toMatchSnapshot()
    })
});