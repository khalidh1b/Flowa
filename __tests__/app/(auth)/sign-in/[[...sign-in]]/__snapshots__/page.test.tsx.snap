// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SignInPage matches snapshot 1`] = `
<DocumentFragment>
  <div>
    mock signin component
  </div>
</DocumentFragment>
`;
